sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVectorUtils.java

```diff
@@ -165,7 +171,17 @@ private static void appendValue(WritableColumnVector dst, DataType t, Object o)
         dst.getChild(1).appendInt(c.days);
         dst.getChild(2).appendLong(c.microseconds);
       } else if (t instanceof DateType) {
-        dst.appendInt(DateTimeUtils.fromJavaDate((Date)o));
+        if (o instanceof Date) {
```

Comment:
@cloud-fan mentioned at
https://github.com/apache/spark/pull/38873#discussion_r1039618680

`ColumnVectorUtils#toBatch` and related `appendValue` methods only used by test cases:

1. this pr add more primitive  dataType support for `toBatch` method and add more test in  `ColumnarBatchSuite#testRandomRows`, do you think this is valuable? I haven't created Jira.

2. `ColumnVectorUtils#toBatch` only used by test, should we move it from `ColumnVectorUtils`  to a test only helper class?

Reply:
We should support all data types, but probably not all java types since it's test-only. We can require tests to only use `Date` or `LocalDate`.

For 2, I think it's fine to keep it here.

Reply:
OK, change to only test `Date` and `Timestamp` due they are default